torch/lib/THD/base/channels/DataChannelTCP.cpp

```diff
@@ -673,6 +673,33 @@ void DataChannelTCP::receive(Scalar& data, int src_rank) {
 }
 
 
+void DataChannelTCP::receive(Tensor& data) {
+  auto request = _receive_worker.push([this, &data]{
+    std::unique_ptr<struct pollfd[]> events(new struct pollfd[this->_processes.size()]);
```

Comment:
It would be good to cache this somewhere. It's small and constant, so it can be reused.